src/components/ContactPage/ContactPage.tsx

```diff
@@ -5,25 +5,28 @@ import {Config} from "../../data/Config";
 
 import {OnMobile, OnMobileTablet, OnDesktop, OnTablet} from "../../helpers/Breakpoints";
 import {formatNumber} from "../../helpers/formatNumber";
+import {translate} from "../../helpers/translate";
 
 import {CloseButton, SubmitButton} from "../Buttons";
 import {ContactPageState} from "./ContactPageState";
 import {ContactForm} from "./ContactForm";
 import {SocialLinks} from "../Layout/Partials/SocialLinks";
 import {Map} from "../Widgets/Map"
+import {LayoutContext, LayoutContextTypes} from "../Layout/LayoutContext";
 
 export class ContactPage extends React.Component<undefined, ContactPageState> {
+    public static readonly contextTypes = LayoutContextTypes;
+    public context: LayoutContext;
 
     public state: ContactPageState = {
         isModalOpen: false,
     };
 
-    public shouldComponentUpdate(nextProps: undefined, nextState: ContactPageState): boolean {
-        return this.state.isModalOpen !== nextState.isModalOpen;
+    public shouldComponentUpdate(nextProps, nextState: ContactPageState, nextContext: LayoutContext): boolean {
```

Comment:
It seems `shouldComponentUpdate` does not receive `nextContext` in React16 https://reactjs.org/docs/react-component.html#shouldcomponentupdate

Reply:
It seems that its big secret if docs does not contain this feature
![image](https://user-images.githubusercontent.com/25794983/31758649-ff75dcea-b4b6-11e7-8e33-fbf82e989746.png)